### PR TITLE
(SERVER-2462) Test v4 Catalog Endpoint

### PR DESCRIPTION
**Pull request by justinstoller**

### CODE REVIEW

**Comment by justinstoller** [comment]
Needs a lot of clean up, and uncommenting/deletion as features land. But afaict it currently, correctly tests what exists (plus what is in my save PR).

**Comment by justinstoller** [comment]
Requires existing PRs at https://github.com/puppetlabs/puppetserver/pull/2000 and https://github.com/puppetlabs/puppetserver/pull/1995 to be merged first.

File: acceptance/suites/tests/catalog/v4_catalog.rb [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+      on(master, "chown -R #{user}: #{userdir_for(user)}")
+
+      on(master, "su -l #{user} -c '/opt/puppetlabs/bin/puppet agent -t'", allow_change)
+      on(master, "su -l #{user} -c '/opt/puppetlabs/bin/puppet agent -t'", allow_change)
+    end
+  end
+
+  step 'create site.pp' do
+    create_remote_file(master, site_path, <<SITEPP)
+# We want to make
```
Comment (Magisus) [inline acceptance/suites/tests/catalog/v4_catalog.rb]: Is this comment unfinished?

**Comment by pcarlisle** [comment]
Yeah a separate pe one is probably good

**Comment by justinstoller** [comment]
Okay... Think this is ready to go!

**Comment by Magisus** [comment]
Later on, once the code is fixed for adding the `transaction_uuid` to the catalog, we should add testing for that here.

**Pull request merged**